src/vdxf/classes/CompactIdAddressObject.ts

```diff
@@ -0,0 +1,153 @@
+
+/**
+ * CompactIdentityObject - Class representing an id in the smallest possible format
+ *
+ * This class is used to represent an identity or address in a compact format, allowing for efficient
+ * storage and transmission. The compact id can be represented either as a fully qualified name (FQN)
+ * or as an identity address (iaddress). The class includes methods for serialization, deserialization,
+ * and validation of the compact id object.
+ *
+ */
+
+import { BN } from 'bn.js';
+import bufferutils from '../../utils/bufferutils';
+import { BigNumber } from '../../utils/types/BigNumber';
+const { BufferReader, BufferWriter } = bufferutils;
+import { SerializableEntity } from '../../utils/types/SerializableEntity';
+import varuint from '../../utils/varuint';
+import { fromBase58Check, toBase58Check, toIAddress } from "../../utils/address";
+import varint from '../../utils/varint';
+import { I_ADDR_VERSION } from '../../constants/vdxf';
+
+export interface CompactIdAddressObjectInterface {
+  version?: BigNumber;
+  flags: BigNumber;
+  address: string;
+  rootSystemName: string;
+}
+
+export interface CompactIdAddressObjectJson {
+  version: number;
+  flags: number;
+  address: string;
+  rootsystemname: string;
+}
+
+export class CompactIdAddressObject implements SerializableEntity {
+
+  static VERSION_INVALID = new BN(0);
+  static FIRST_VERSION = new BN(1);
+  static LAST_VERSION = new BN(1);
+  static DEFAULT_VERSION = new BN(1);
+
+  static IS_FQN = new BN(1);
+  static IS_IDENTITYID = new BN(2);
+
+  version: BigNumber;
+  flags: BigNumber;
+  address: string;
+  rootSystemName: string;
+
+  constructor(data?: CompactIdAddressObjectInterface) {
+    this.version = data?.version || new BN(CompactIdAddressObject.DEFAULT_VERSION);
+    this.flags = data?.flags || new BN(0);
+    this.address = data?.address || '';
+    this.rootSystemName = data?.rootSystemName || 'VRSC';
+  }
+
+  isFQN(): boolean {
```

Comment:
This should by type, not flags, because it can't be both fqn and i-addr at the same time